### PR TITLE
Remove ancient streams migration guide

### PR DESCRIPTION
**Pull request by raboof**

### CODE REVIEW

**Review by johanandren — APPROVED** [review]

File: akka-docs/src/main/paradox/project/migration-guides.md [hunk 1]
```diff
@@ -11,8 +11,7 @@
 * [migration-guide-persistence-experimental-2.3.x-2.4.x](migration-guide-persistence-experimental-2.3.x-2.4.x.md)
```
Comment (patriknw) [inline akka-docs/src/main/paradox/project/migration-guides.md]: this one is probably just as old, or older

File: akka-docs/src/main/paradox/project/migration-guides.md [hunk 2]
```diff
@@ -11,8 +11,7 @@
...
 * [migration-guide-eventsourced-2.3.x](migration-guide-eventsourced-2.3.x.md)
 * [migration-guide-2.3.x-2.4.x](migration-guide-2.3.x-2.4.x.md)
-* [migration-guide-stream-1.0-2.x](migration-guide-stream-1.0-2.x.md)
```
Comment (patriknw) [inline akka-docs/src/main/paradox/project/migration-guides.md]: is the reason that the page migration-guide-stream-1.0-2.x.md contains links that are not valid any more?
by removing the page we might introduce a 404 from elsewhere such as some old blog post

**Review by patriknw — COMMENTED** [review]
I don't mind cleaning up this but then we should maybe cleanup more old stuff. It contains references to all old stuff (which is pretty cool, but maybe not very useful any more).

File: akka-docs/src/main/paradox/project/migration-guides.md (hunk 2 above)
Comment (raboof) [reply akka-docs/src/main/paradox/project/migration-guides.md]: That was the trigger, yes: we removed `akka-stream-and-http-experimental` entirely so now the links to it 404. I'm not sure there are any links to here (I'd expect them to go directly to `akka-stream-and-http-experimental`) but if so we should fix those as well I think. Perhaps just fix them as they pop up in reports?

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**